onnx/shape_inference/implementation.cc

```diff
@@ -292,11 +292,10 @@ static void InferShapesImpl(
               function_opset_imports[opset_import.domain()] = static_cast<int>(opset_import.version());
             } else {
               if (it->second != opset_import.version()) {
-                throw ONNX_NAMESPACE::InferenceError(
-                    "ONNX models do not support multiple opset version imports for a domain. Function " +
-                            schema->Name() + " imports opset version " + std::to_string(opset_import.version()) +
-                            " for domain " + (opset_import.domain().empty() ? "ai.onnx" : opset_import.domain()) +
-                            " where as the model imports opset version " + std::to_string(it->second));
+                fail_shape_inference("ONNX models don't support multiple opset version imports for a domain. Function ",
```

Comment:
Is it a "shape" inference? I thought using ONNX_NAMESPACE::InferenceError is more proper

Reply:
fail_shape_inference throws a InferenceError... not sure what you mean

Reply:
Yes eventually they will throw the same error. Just thought this one is actually not a "shape" inference error so directly using `fail_shape_inference` looks a little weird to me.